apps/mosaico/src/Mosaico.purs

```diff
@@ -66,54 +70,87 @@ routes :: Match MosaicoPage
 routes =
   articleRoute <|> frontpageRoute
 
-app :: Component {}
+app :: Component Props
 app = do
+  initialValues <- getInitialValues
+  component "Mosaico" $ mosaicoComponent initialValues
+
+mosaicoComponent
+  :: InitialValues
+  -> Props
+  -> Render Unit (UseEffect MosaicoPage (UseEffect Unit (UseState State Unit))) JSX
+mosaicoComponent initialValues props = React.do
+  state /\ setState <- useState initialValues.state { article = props.article }
+
+  -- Listen for route changes and set state accordingly
+  useEffectOnce $ locations (routeListener setState) initialValues.nav
+  useEffect state.route do
+    case state.route of
+      Frontpage -> do
+        if null state.articleList
+        then Aff.launchAff_ do
+          frontpage <- Lettera.getFrontpage HBL
+          liftEffect $ setState \s -> s { articleList = frontpage, article = Nothing }
+        -- Set article to Nothing to prevent flickering of old article
+        else liftEffect $ setState \s -> s { article = Nothing }
```

Comment:
Isn't this in `Effect` already? Seems like no-op unless I'm missing something.

Reply:
Yep, you're right.